src/aws_signature.erl

```diff
@@ -273,12 +281,19 @@ sign_v4_query_params(AccessKeyID, SecretAccessKey, Region, Service, DateTime, Me
     FinalQueryParams1 =
         add_credential_query_param(FinalQueryParams0, CredentialScope, AccessKeyID),
     FinalQueryParams2 = maybe_add_session_token_query_param(FinalQueryParams1, SessionToken),
+    FinalQueryParams3 = maybe_add_tag_query_param(FinalQueryParams2, Tag),
 
-    FinalQueryParams = add_date_header(FinalQueryParams2, LongDate),
+    FinalQueryParams = add_date_header(FinalQueryParams3, LongDate),
     HostHeader = host_header_from_url(URLMap),
+    Headers =
+        if Tag == undefined ->
+                [HostHeader];
+           Tag =/= undefined ->
+                [HostHeader, {<<"X-Amz-Tagging">>, hackney_url:urldecode(Tag)}]
```

Comment:
We should accept an unescaped value instead, the same that the caller would use for the header.

Reply:
But the session_token in Options is escaped here: https://github.com/aws-beam/aws-erlang/blob/master/src/aws_s3_presigned_url.erl#L44C21-L45C66

Reply:
Correct, that's because we put it in the query params. My point is that we should not add tag to the query params, and only add it to headers for signing. So given that, the value should match the actual header value.